src/Sarif/Comparers/ComparerHelpers.cs

```diff
@@ -0,0 +1,205 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif.Comparers
+{
+    internal class ComparerHelper
+    {
+        /// <summary>
+        /// Compare 2 object refences. Return value false presents need to
+        /// compare objects values to get final result.
+        /// </summary>
+        /// <param name="left">The first object to compare.</param>
+        /// <param name="right">The second object to compare.</param>
+        /// <param name="result">
+        /// 0 if both objects are same or both are null.
+        /// -1 if the first object is null and the second object is not null.
+        /// 1 if the first object is not null and the second object is null.
+        /// </param>
+        /// <returns>Return true if can get a definite compare result, otherwise return false.</returns>
```

Comment:
how is this getting compiled? where? who can see your comment outside this source file?

otherwise convert this comment to a regular comment and make sure it's useful.

Reply:
IDE intellisense shows this comment for people wants to call it. I d like to keep the comment for this method since it has some behaviors cannot tell by just signatures. 
Removed comments for other methods in this file.

Reply:
changed to normal comments since its internal class its no visible outside of Sarif.Sdk